fineract-provider/src/main/java/org/apache/fineract/useradministration/service/AppUserWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -216,6 +217,28 @@ public CommandProcessingResult updateUser(final Long userId, final JsonCommand c
             	clients = this.clientRepositoryWrapper.findAll(clientIds);
             }
 
+            final String passwordParamName = "password";
+            final String currentPasswordParamName = "currentPassword";
+            final String currentPasswordEncodedValue = command.passwordValueOfParameterNamed(currentPasswordParamName, platformPasswordEncoder,
+                        userToUpdate.getId());
+            AppUser currentUser = getAppUserIfPresent();
+
+            if (command.hasParameter(passwordParamName)) {
+            for(String id : currentUser.getRolesAsIdStringArray()) {
+                if (!id.equals("1")){
```

Comment:
I'm not familiar enough with the code to know what a Role as ID = "1" means... perhaps an inline comment could help to clarify this to future readers of this code?